Samples.Swift/Samples.Swift/ViewControllers/CheckoutViewController/CheckoutViewController.swift

```diff
@@ -0,0 +1,478 @@
+      items[index] = item
+    }
+    else {
+      items.append(item)
+    }
+    tableView.reloadData()
+  }
+
+  deinit { }
+}
+
+extension CheckoutViewController: UITableViewDelegate, UITableViewDataSource {
+
+  func numberOfSections(in tableView: UITableView) -> Int {
+      return 2
+  }
+
+  func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+    switch section {
+    case 0:
+      return "Items"
+
+    case 1:
+      return "Cart total"
+
+    default:
+      return ""
+    }
+  }
+
+  func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    if indexPath.section == 0 {
+      if indexPath.row >= items.count {
+        guard
+          let cell = tableView.dequeueReusableCell(withIdentifier: "AddItemCell", for: indexPath) as? AddItemCell
+          else { return UITableViewCell() }
+        return cell
+      }
+      else {
+        guard
+          let cell = tableView.dequeueReusableCell(withIdentifier: "ItemCell", for: indexPath) as? ItemCell
+          else { return UITableViewCell() }
+        cell.setupUIWithItem(withItem: items[indexPath.row])
+        return cell
+      }
+    }
+    else {
+      guard
+        let cell = tableView.dequeueReusableCell(withIdentifier: "TotalCell", for: indexPath) as? TotalCell
+        else { return UITableViewCell() }
+      cell.setupUIWithTotal(withSubtotal: "$\(getItemTotal())", withTax: "$\(getTaxTotal())", withTotal: "$\(getTotal())")
+      return cell
+    }
+  }
+
+  func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+    if indexPath.section == 0 {
+      if indexPath.row >= items.count {
+        return 48
+      }
+      return UITableView.automaticDimension
+    }
+    else  {
+      return UITableView.automaticDimension
+    }
+  }
+
+  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    if section == 0 {
+      return items.count + 1
+    }
+    else {
+      return 1
+    }
+  }
+
+  func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+    if indexPath.section == 0 {
+      if indexPath.row >= items.count {
+        let addItemViewController = AddItemViewController()
+        addItemViewController.modalPresentationStyle = .overFullScreen
+        addItemViewController.delegate = self
+        addItemViewController.index = indexPath.row
+        present(addItemViewController, animated: true)
+        tableView.deselectRow(at: indexPath, animated: true)
+      }
+      else {
+        let addItemViewController = AddItemViewController(item: items[indexPath.row])
+
+        addItemViewController.modalPresentationStyle = .overFullScreen
+        addItemViewController.delegate = self
+        addItemViewController.index = indexPath.row
+
+        present(addItemViewController, animated: true)
+        tableView.deselectRow(at: indexPath, animated: true)
+      }
+    }
```

Comment:
good usecase for `guard`, to avoid an extra level of indentation: 
```swift
guard indexPath.section == 0 else {
  return 
}
 <the rest of the function>
```